### PR TITLE
Better handling of Unit tests in D and DevPreview

### PR DESCRIPTION
**Pull request by hrcornejo**
- The execution status of all Unit tests, for all components, are reported into Jira.
- The DTR document is generated and contains information that is consistent with Jira.
- When doing a Developer Preview, the Release Manager pipeline fails only at the end of the Finalize Stage.
- When deploying to DEV, the Release Manager pipeline fails at the end of the Build Stage.
- The Release Manager reports the the failure status into its Release Status issue.

### CODE REVIEW

**Comment by metmajer** [comment]
It would be nice, but I am not sure if this implementation will make the requirements. Consider that a component might decide to fail the build after the first failing unit tests. Currently, components are executed in groups with a `failFast` property. Therefore, after one failing component, the release manager pipeline would fail.

**Comment by hrcornejo** [comment]
> I see many many files in which a generic value is changed by sergio.sacristan%40domain.com
> 
> Is it on purpose?
@victorpablosceruelo there are only 3 files changed between this branch and master.

**Comment by hrcornejo** [comment]
> It would be nice, but I am not sure if this implementation will make the requirements. Consider that a component might decide to fail the build after the first failing unit tests. Currently, components are executed in groups with a `failFast` property. Therefore, after one failing component, the release manager pipeline would fail.

@metmajer added failFast dependent of WIP.
With 2 projects in parallel with this configuration:
image
Launching a WIP all the stages are executed, doc generated and issues updated in Jenkins (with failing tests in one component):
image
Launching a 1.1 version to DEV, the pipeline fails after Build:
image

**Comment by metmajer** [comment]
@hrcornejo fantastic! Interestingly, in the second example, the pipeline fails, but we don't see that the component failed. I assume this is because the component itself does not throw an exception/error on test failure?

File: test/groovy/org/ods/orchestration/BuildStageSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    def setup() {
+        script = new PipelineScript()
+        steps = Mock(PipelineSteps)
+        levaDocScheduler = Mock(LeVADocumentScheduler)
+        project = Spy(createProject())
+        util = Mock(MROPipelineUtil)
+        jira = Mock(JiraUseCase)
+        junit = Mock(JUnitTestReportsUseCase)
+        jenkins = Mock(JenkinsService)
+        logger = new Logger(script, !!script.env.DEBUG)
```
Comment (s2oBCN) [inline test/groovy/org/ods/orchestration/BuildStageSpec.groovy:42]: This is a test, why initialize with !!script.env.DEBUG?

**Review by s2oBCN — DISMISSED** [review]

File: test/groovy/org/ods/orchestration/BuildStageSpec.groovy (hunk 1 above)
Comment (hrcornejo) [reply test/groovy/org/ods/orchestration/BuildStageSpec.groovy]: C&P from other test. Updated in new commit

**Review by s2oBCN — APPROVED** [review]

**Pull request merged**